### PR TITLE
initial whl workflow

### PR DESCRIPTION
**Pull request by andy-neuma**
SUMMARY:
* add helper script for building in python venv
* add GHA action for building whl file
* add helper script for converting python version to whl style
* add callable workflow to use for constructing whl files

TEST PLAN:
this is a seed PR ... testing will commence once these changes are merged

### CODE REVIEW

File: .github/actions/nm-build-vllm-whl/action.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      VENV="${{ inputs.venv }}-${COMMIT:0:7}"
+      source $(pyenv root)/versions/${{ inputs.python }}/envs/${VENV}/bin/activate
+      SUCCESS=0
+      pip3 wheel --no-deps -w dist . || SUCCESS=$?
+      echo "status=${SUCCESS}" >> "$GITHUB_OUTPUT"
+      exit ${SUCCESS}
+      BASE=$(./.github/scripts/convert-version ${{ inputs.python }})
+      WHL_FILEPATH=$(find dist -iname "*nm_vllm*${BASE}*.whl")
+      WHL=$(basename ${WHL_FILEPATH})
+      echo "whl=${WHL}" >> "$GITHUB_OUTPUT"
```
Comment (bnellnm) [inline .github/actions/nm-build-vllm-whl/action.yml:32]: does any of this stuff get executed?  there's an `exit` right before this section of code

File: .github/scripts/build [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    echo "usage: ${0} <options>"
+    echo
+    echo "  -a    - pypi server address (optional)"
+    echo "  -p    - python version"
+    echo "  -v    - name for virtualenv"
+    echo "  -h    - this list of options"
+    echo
+}
+
+PYPI_IP=192.168.201.226
```
Comment (bnellnm) [inline .github/scripts/build:16]: is this script going to end up in the public repo?  not a big deal for now but we might want to remove this ip addr.

**Review by dhuangnm — APPROVED** [review]

File: .github/scripts/build (hunk 2 above)
Comment (dhuangnm) [reply .github/scripts/build:16]: maybe we can add it as a secret to GHA?

Comment (andy-neuma) [reply .github/scripts/build:16]: yeah, agree ... i'll update and remove it ...

**Pull request merged**